models/migrations/v58.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/go-sql-driver/mysql"
+	"github.com/go-xorm/xorm"
+)
+
+func mysqlColumnsToUTF8MB4(x *xorm.Engine) (err error) {
+	if !setting.UseMySQL {
+		log.Info("Nothing to do")
+		return nil
+	}
+
+	const maxvc = 191
+	migrationSuccess := true
+
+	tables, err := x.DBMetas()
+	if err != nil {
+		return fmt.Errorf("cannot get tables: %v", err)
+	}
+	for _, table := range tables {
+		readyForConversion := true
+		for ix := range table.Indexes {
+			col := table.GetColumn(ix)
+			if col == nil {
+				continue
+			}
+			if !(col.SQLType.Name == "VARCHAR" && col.Length > maxvc) {
+				continue
+			}
+			log.Info("reducing column %s.%s from %d to %d bytes", table.Name, ix, col.Length, maxvc)
+			var sqlstmt = fmt.Sprintf("alter table `%s` change column `%s` `%s` varchar(%d)", table.Name, ix, ix, maxvc)
```

Comment:
Oops, something that slipped through the cracks: this one should be `:=` as well.